### ISSUE TITLE
[subplat] Trim whitespace on Stripe metadata before validation

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#11932]
We had a functional test CI failure that was affecting pretty much all branches between yesterday and today that ended up being due to a newline character at the end of a metadata value[1].

While Stripe metadata expansion work (specifically FXA-4165) will fix this ultimately, Lauren had the good idea to trim the whitespace before validation in the meantime.

[1]: [Slack thread|https://mozilla.slack.com/archives/CLV3KMZ8B/p1644529933612559]
TL;DR root cause for CI failures: a DEV product (same env used for our functional tests) was failing the response validation for the /plans endpoint on the front-end due to the `webIconURL` metadata value having a `/n` at the end. This value was entered in through the Stripe dashboard which lets you enter anything in there.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/FXA-4606)
